src/networking.c

```diff
@@ -1590,42 +1590,91 @@ int _writeToClient(client *c, ssize_t *nwritten) {
         return C_OK;
     }
 
-    if (c->bufpos > 0) {
-        *nwritten = connWrite(c->conn,c->buf+c->sentlen,c->bufpos-c->sentlen);
-        if (*nwritten <= 0) return C_ERR;
-        c->sentlen += *nwritten;
+    int nbuf = listLength(c->reply);
+    /* When the reply list is not empty, it's better to use writev to save us some
+     * system calls and TCP packets. */
+    if (nbuf > 0) {
+        if (c->bufpos > 0) ++nbuf; /* include the static reply buffer. */
+        struct iovec iov[nbuf > IOV_MAX ? IOV_MAX : nbuf];
+        int iovcnt = 0;
+        size_t cum_bytes = 0;
```

Comment:
```suggestion
        size_t iovec_len = 0;
```

Reply:
You are right. Maybe `iov_len_sum`?

Reply:
What I have in mind is `iov_bytes_len`, is that ok?

Reply:
Done.